src/main/java/subway/service/LineService.java

```diff
@@ -0,0 +1,71 @@
+package subway.service;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import subway.domain.Station;
+import subway.domain.Line;
+import subway.domain.repository.LineRepository;
+import subway.dtos.request.LineRequest;
+import subway.dtos.response.LineResponse;
+
+import javax.persistence.EntityManager;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional(readOnly = true)
+public class LineService {
+
+    private LineRepository lineRepository;
+    private EntityManager entityManager;
+
+    public LineService(LineRepository lineRepository, EntityManager entityManager){
+        this.lineRepository = lineRepository;
+        this.entityManager = entityManager;
+    }
+
+    @Transactional
+    public LineResponse saveLine(LineRequest lineRequest) {
+        Station upStation = entityManager.find(Station.class, lineRequest.getUpStationId());
+        Station downStation = entityManager.find(Station.class, lineRequest.getDownStationId());
+        Line line = lineRepository.save(new Line(
+                lineRequest.getName(),
+                lineRequest.getColor(),
+                upStation,
+                downStation
+                ));
+        return createLineResponse(line);
+    }
+
+    public List<LineResponse> findAllLines() {
+        return lineRepository.findAll().stream()
+                .map(this::createLineResponse)
+                .collect(Collectors.toList());
+    }
+
+    public LineResponse findLineById(Long id) {
+        Line line = lineRepository.findById(id).get();
+        return createLineResponse(line);
+    }
+
+    @Transactional
+    public void updateLine(Long id, LineRequest lineRequest) {
+        Line line = lineRepository.findById(id).get();
+        line.updateLine(lineRequest.getName(), lineRequest.getColor());
+        lineRepository.save(line);
```

Comment:
```suggestion
        line.updateLine(lineRequest.getName(), lineRequest.getColor());
```
 lineRepository.save(line)가 없으면 어떻게 될까요~?

Reply:
더티 체킹을 통해서 업데이트가 자동으로 되는군요.. 신기하네요! update가 save로 가능하다고는 들었는데 또 새로운 걸 배워갑니다!